packages/tasks/src/model-libraries-snippets.ts

```diff
@@ -82,6 +82,43 @@ export const bm25s = (model: ModelData): string[] => [
 retriever = BM25HF.load_from_hub("${model.id}")`,
 ];
 
+export const depth_anything_v2 = (): string[] => {
+	const shellCommands = [
+		`# Install the Depth-Anything-V2 library
+git clone https://github.com/DepthAnything/Depth-Anything-V2
+cd Depth-Anything-V2
+pip install -r requirements.txt`,
+	];
+
+	const pythonCode = [
+		`# Load the model and infer depth from an image
+import cv2
+import torch
+
+from depth_anything_v2.dpt import DepthAnythingV2
+
+DEVICE = 'cuda' if torch.cuda.is_available() else 'mps' if torch.backends.mps.is_available() else 'cpu'
+
+model_configs = {
+    'vits': {'encoder': 'vits', 'features': 64, 'out_channels': [48, 96, 192, 384]},
+    'vitb': {'encoder': 'vitb', 'features': 128, 'out_channels': [96, 192, 384, 768]},
+    'vitl': {'encoder': 'vitl', 'features': 256, 'out_channels': [256, 512, 1024, 1024]},
```

Comment:
Maybe let's go with a single configuration and point people to pick which one to use

Reply:
(nit) I was thinking of using the `modelId` to know which encoder to use and therefore generate the snippet only with the needed config. With the current snippet, users will see all config displayed even if only one is needed.

Reply:
Yes so the problem is that, as the author isn't using the mixin, we need those 10 lines of code to first instantiate the appropriate neural network architecture before loading the weights. There's not really a way around that..

Reply:
The mixin is not the factor here. The recommendation is to do

```
if (model.id === "model1) {
   /// add code specific to first model
} else if
```

Then the final snippet could have model-specific code, such as:

```
import cv2
import torch
from depth_anything_v2.dpt import DepthAnythingV2

model = DepthAnythingV2(encoder='vits', features=64, out_channels=[48, 96, 192, 384])
filepath = hf_hub_download(repo_id="${model.id}", filename="depth_anything_v2_vits.pth
")
state_dict = torch.load(filepath, map_location="cpu")
model.load_state_dict(state_dict).eval()

raw_img = cv2.imread("your/image/path")
depth = model.infer_image(raw_img) # HxW raw depth map in numpy`
```

Reply:
Ok I pushed a commit, let me know what you think :)

Reply:
Nice :fire: